### PR TITLE
Update mmr

### PR DESCRIPTION
**Pull request by jiqiang90**
-  https://github.com/subquery/merkle-mountain-range/pull/2 merge and release, to resolve inconsistent mmr value issue
-  changed offset value to the 1st indexing opertaion block height -1
-  remove unnecessary steps in appendMmr, as this is steps is duplicate in ensureStartHeight
-  testing docker file https://gist.github.com/jiqiang90/dfc4210c6d0c68f42bdbc232c7a05ddb

### CODE REVIEW

**Comment by jiqiang90** [comment]
Closes #855

File: packages/node/src/indexer/indexer.manager.ts [hunk 1]
```diff
@@ -170,6 +170,14 @@ export class IndexerManager {
     this.fetchService.latestProcessed(block.block.header.number.toNumber());
     this.prevSpecVersion = block.specVersion;
     if (this.nodeConfig.proofOfIndex) {
+      if (this.blockOffset === undefined) {
```
Comment (ianhe8x) [inline packages/node/src/indexer/indexer.manager.ts:173]: I can see a reason you move the blockOffset initialization here is to skip no operation blocks. If so, the better place is to move it to line 162 and attach it to tx.

File: packages/node/src/indexer/mmr.service.ts [hunk 2]
```diff
@@ -52,54 +52,71 @@ export class MmrService implements OnApplicationShutdown {
...
 
-  async fetchBlockOffsetFromDb(): Promise<number | null> {
-    const blockOffset = await this.metadataRepo.findOne({
-      where: { key: 'blockOffset' },
-    });
-    if (blockOffset === null) {
-      throw new Error(`Poi service failed to fetch block offset from metadata`);
+  // await index 1st block and set the blockOffset
+  async awaitFetchBlockOffsetFromDb(): Promise<number> {
+    while (!this.isShutdown && this.blockOffset === null) {
```
Comment (ianhe8x) [inline packages/node/src/indexer/mmr.service.ts:111]: don't get why use `while` here

**Review by ianhe8x — CHANGES_REQUESTED** [review]
there're redundant codes can be optimised

File: packages/node/src/indexer/indexer.manager.ts [hunk 3]
```diff
@@ -214,13 +226,17 @@ export class IndexerManager {
...
     this.filteredDataSources = this.filterDataSources(startHeight);
     this.fetchService.register((block) => this.indexBlock(block));
+  }
 
-    if (this.nodeConfig.proofOfIndex) {
-      void this.mmrService.syncFileBaseFromPoi().catch((err) => {
+  private setBlockOffset(offset: number): void {
+    this.blockOffset = offset;
+    logger.info(`set blockoffset to ${offset}`);
+    void this.mmrService
```
Comment (stwiname) [inline packages/node/src/indexer/indexer.manager.ts:234]: What's the reason for not awaiting this?

Comment (jiqiang90) [reply packages/node/src/indexer/indexer.manager.ts:234]: I assume you are asking  the `mmr.syncFileBaseFromPoi` here, while sync the between poi and filebased mmr,  this should be independent and we try to not block indexing service.

**Review by stwiname — APPROVED** [review]

**Review by ianhe8x — APPROVED** [review]

**Pull request merged**